libs/vkd3d/resource.c

```diff
@@ -5838,6 +5838,22 @@ static void vkd3d_create_buffer_srv(vkd3d_cpu_descriptor_va_t desc_va,
         VK_CALL(vkUpdateDescriptorSets(device->vk_device, vk_write_count, vk_write, 0, NULL));
 }
 
+static void vkd3d_texture_view_desc_fixup(struct d3d12_device *device, struct vkd3d_texture_view_desc *desc)
+{
+    if (device->device_info.properties2.properties.vendorID == VKD3D_VENDOR_ID_NVIDIA)
+    {
+        /* D3D allows some reinterpretation between Texture2D and Texture2DArray.
+         * Texture2D in shader can read a resource with 1 array layer,
+         * and Texture2DArray can read a Texture2D descriptor.
+         * NVIDIA does not correctly deal with Texture2DArray unless we always emit 2D_ARRAY views.
+         * Other implementations don't seem to care, so just emit the natural 2D view. */
```

Comment:
I'd feel safer with a `WARN` or something here reminding us that this workaround exists in case things break in the future, other than that, looks... good I guess. Not a fan of hacks like these but it's not like we can do anything here without potentially sacrificing perf.

Reply:
Added.